### DISCUSSION TITLE
The influence of humidity on the mass fraction of other AIR components

### DISCUSSION THREAD

**Discussion by mst-parkhachev** [firemodels/fds#13500] [General]
Dear colleagues!

I could not find a detailed description in the manual of how exactly the HUMIDITY parameter affects the mass fraction of other air components at the beginning of the calculation.
I found the following comment by dr_jfloyd
https://groups.google.com/g/fds-smv/c/liYnTJCVEWA/m/dH_oWwh4AQAJ

> Y_CO2_INFTY is the dry (no water vapor) mass fraction of CO2. FDS first defines the dry composition of air based on Y_O2_INFTY and Y_CO2_INFTY. Then FDS adds in water vapor based on the HUMIDITY.  The default HUMIDTY will result in about 0.5 % water vapor which when added will slightly drop the CO2 mass fraction. 

Air in FDS by default consists of four components: oxygen, carbon dioxide, nitrogen and water vapor. Is it true that Y_O2_INFTY and Y_CO2_INFTY parameters are used to calculate the mass fractions of **oxygen**, **carbon dioxide** and **nitrogen** assuming that there is no water vapor at all? Is it true that their mass fractions are then **proportionally** reduced to account for the presence of water according to the HUMIDTY parameter?

Could you also provide a link to the FDS code where this calculation is made?

I found the functions WATER_VAPOR_MASS_FRACTION(HUMIDITY,TEMP,PZONE) and RELATIVE_HUMIDITY(Y_H2O,TEMP,PZONE), but they do not answer my question.

**Comment by drjfloyd** [comment]
Yes, the INFTY inputs are for dry air (no moisture).  The adjustment is just done by simple algebra.  If you had 1 kg of dry air being 0.77 kg of N2 and 0.23 kg of O2 and your humidity was a 2 % mass fraction, then that would mean you now have 0.77 kg N2, 0.23 kg O2, and x kg of H2O where x satisfies  x/(1+x) = 0.02.  Once you know x you can renormalize everthing back to 1, i.e., N2 becomes 0.77/(1+x).

The default species mixutures are defined in read.f90 in SETUP_PREDEFINED_SMIX.

**Comment by mst-parkhachev** [comment]
Thank You for detailed answer, Dr. Jason!
